packages/material-ui/src/useAutocomplete/useAutocomplete.js

```diff
@@ -129,6 +129,7 @@ export default function useAutocomplete(props) {
   const [focusedTag, setFocusedTag] = React.useState(-1);
   const defaultHighlighted = autoHighlight ? 0 : -1;
   const highlightedIndexRef = React.useRef(defaultHighlighted);
+  const [focusedIndex, setFocusedIndex] = React.useState(defaultHighlighted);
```

Comment:
It's too slow to rerender for this state. AFAIK

```suggestion
```

Reply:
Can we improve performance for large options using React.useMemo()? 

https://github.com/mui-org/material-ui/pull/23297/files#diff-2f1d031d3019355c18f9a0f7bcdc7bf48668ce73fe30454ec44aa6d0c3db562cR411-R428

Reply:
Maybe, virtualization is one case, we also need to consider when people use the render option API with "heavy" components, like the `MenuItem`.